addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/internal/OneWireGPIOHandlerFactory.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.onewiregpio.internal;
+
+import static org.openhab.binding.onewiregpio.OneWireGPIOBindingConstants.THING_TYPE;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.onewiregpio.handler.OneWireGPIOHandler;
+import org.osgi.service.component.ComponentContext;
+
+/**
+ * The {@link OneWireGPIOHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Anatol Ogorek - Initial contribution
+ */
+public class OneWireGPIOHandlerFactory extends BaseThingHandlerFactory {
```

Comment:
Add `@Component(service = ThingHandlerFactory.class, immediate = true)`  here to auto generate the DS XML description (see corresponding comment there).